examples/image_list/test/load_test.dart

```diff
@@ -0,0 +1,553 @@
+  0xe3, 0x02, 0xbd, 0xa8, 0x57, 0x6b, 0xa4, 0x3b, 0x9b, 0x6a, 0x9a, 0x31, 0x83, 0xce, 0x0b, 0xfc,
+  0x4a, 0x1d, 0xbb, 0xb4, 0xca, 0x8b, 0x93, 0x63, 0xf9, 0x6e, 0x43, 0x5e, 0x53, 0xcb, 0x5d, 0xa8,
+  0x0f, 0xa6, 0x25, 0x5a, 0x8d, 0x90, 0x3f, 0xfd, 0xfe, 0x23, 0x9a, 0x15, 0x8f, 0xc3, 0xa3, 0x13,
+  0xc6, 0x99, 0xcf, 0x45, 0x6f, 0xc6, 0xd7, 0x87, 0x8f, 0xf8, 0xe5, 0x2f, 0x7e, 0x85, 0xf5, 0xbb,
+  0xdf, 0x7c, 0x50, 0x44, 0xaa, 0x68, 0xb3, 0xd5, 0x60, 0x73, 0xbd, 0xc1, 0x78, 0x3c, 0xe6, 0xab,
+  0x7b, 0x4f, 0x74, 0xbf, 0xc6, 0xfa, 0x6a, 0x80, 0xef, 0x97, 0xd4, 0x5d, 0x22, 0x36, 0xaf, 0xad,
+  0xb2, 0xae, 0xa2, 0x19, 0xeb, 0xf8, 0xda, 0xbf, 0x91, 0xb6, 0x63, 0x9e, 0xa9, 0xd1, 0xea, 0xad,
+  0x59, 0x6b, 0xb6, 0xf9, 0xef, 0xf1, 0x21, 0xbd, 0xd1, 0x39, 0xff, 0x07, 0xb3, 0x0e, 0x83, 0x72,
+  0xca, 0x6e, 0x25, 0x31, 0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82,
+];
+
+HttpClient sampleHttpProvider() {
+  final HttpClient httpClient = MockHttpClient();
+  final List<Uint8List> chunks = <Uint8List> [Uint8List.fromList(kSamplePng)];
+  final MockHttpClientRequest request = MockHttpClientRequest();
+  final MockHttpClientResponse response = MockHttpClientResponse();
+  when(httpClient.getUrl(any)).thenAnswer((_) => Future<HttpClientRequest>.value(request));
+  when(request.close()).thenAnswer((_) => Future<HttpClientResponse>.value(response));
+  when(response.statusCode).thenReturn(HttpStatus.ok);
+  when(response.contentLength).thenReturn(kSamplePng.length);
+  when(response.listen(
+    any,
+    onDone: anyNamed('onDone'),
+    onError: anyNamed('onError'),
+    cancelOnError: anyNamed('cancelOnError'),
+  )).thenAnswer((Invocation invocation) {
+    final void Function(List<int>) onData = invocation.positionalArguments[0];
+    final void Function(Object) onError = invocation.namedArguments[#onError];
+    final void Function() onDone = invocation.namedArguments[#onDone];
+    final bool cancelOnError = invocation.namedArguments[#cancelOnError];
+
+    return Stream<Uint8List>.fromIterable(chunks).listen(
+      onData,
+      onDone: onDone,
+      onError: onError,
+      cancelOnError: cancelOnError,
+    );
+  });
+
+  return httpClient;
+}
+
+Future<void> main() async {
+  group(NetworkImage, ()
+  {
+    testWidgets('Ensure images are loaded', (WidgetTester tester) async {
+      debugNetworkImageHttpClientProvider = sampleHttpProvider;
+
+      await tester.pumpWidget(const MyApp(0));
+
+      final Finder allImages = find.byType(Image);
+      expect(allImages, findsNWidgets(50));
+
+      for (Element e in allImages.evaluate()) {
+        await tester.runAsync(() async {
+          final Image image = e.widget;
+          await precacheImage(image.image, e);
+        });
+      }
+
+      await tester.pumpAndSettle();
+      await expectLater(
+        find.byType(MyHomePage),
+        matchesGoldenFile(
+          'load.widgets_loaded.done.png',
```

Comment:
This should probably use the flutter-specific golden matcher that matches golden files in the golden repository or in the new Flutter Gold service.

/cc @Piinks

Reply:
This looks like this is for an example app. Is this intended to illustrate to developers how to use golden file testing? 
Or is it meant to be used to test against framework changes in flutter/flutter?

Reply:
In my mind former, rather than latter. This is an example app that uses some loaded images that shows how to use golden files to test it(getting images loaded for golden screen testing seems to be tricky thing).

Reply:
Ian suggested that having goldens in flutter repo is not a good idea, so based on that I have moved goldens for this example into flutter/goldens repo.

Reply:
I am curious to see how this works then. 

Does this example use the Flutter-repo golden file comparator? If it does not, it will not compare to files in the goldens repository. 

Is this example meant to be run by developers to see how golden file testing works? If they run the example on their own it may not work.

If this is a sample application, maybe the samples repo would be a better place for it where you could store image files?

Reply:
Yes now this test uses flutter-repo golden file comparator(configured by  examples/image_list/test/flutter_test_config.dart that I added).

Reply:
Can we add documentation explaining this exception? Since this is meant to be an example to learn from, I don't want folks to think this is how they can do golden tests. The flutter_goldens package is only for testing flutter golden files.

Reply:
If I understand correctly, there is nothing exceptional in this code that makes it flutter_goldens specific. `expectLater(...matchesGoldenFile(...))` remains the same whether you use flutter_goldens or your own goldens in your `test` folder, right?

Reply:
Apologies, I did not mean specifically here on this line of code. 🙂 

I meant where you have added the flutter_goldens package to the pubspec, or in the flutter_test_config you are adding. If someone were to take this example, and run it themselves in a fresh project, it would not work unless they have the actual file present in the test folder. Furthermore, if someone uses this as a guide and sets up their tests to use the flutter_golden package for their own app, it will not work. 
My only aim is to avoid confusion, so I thought making mention of this would be worthwhile.

Reply:
Okay, added comment into flutter_test_config.dart.